homeassistant/components/knx/time.py

```diff
@@ -0,0 +1,103 @@
+"""Support for KNX/IP time."""
+from __future__ import annotations
+
+from datetime import time as dt_time
+import time as time_time
+from typing import Final
+
+from xknx import XKNX
+from xknx.devices import DateTime as XknxDateTime
+
+from homeassistant import config_entries
+from homeassistant.components.time import TimeEntity
+from homeassistant.const import (
+    CONF_ENTITY_CATEGORY,
+    CONF_NAME,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CONF_RESPOND_TO_READ,
+    CONF_STATE_ADDRESS,
+    CONF_SYNC_STATE,
+    DATA_KNX_CONFIG,
+    DOMAIN,
+    KNX_ADDRESS,
+)
+from .knx_entity import KnxEntity
+
+_TIME_TRANSLATION_FORMAT: Final = "%H:%M:%S"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up entities for KNX platform."""
+    xknx: XKNX = hass.data[DOMAIN].xknx
+    config: list[ConfigType] = hass.data[DATA_KNX_CONFIG][Platform.TIME]
```

Comment:
Why isn't config a part of `hass.data[DOMAIN]` ?

Reply:
Tbh I don't remember. I can probably refactor that so it lives in the KNXModule class in a Followup PR (this is used like that in every platform).